src/hw1/bash_imitating_functions.py

```diff
@@ -0,0 +1,58 @@
+def _read_file(filename):
+    try:
+        file = open(filename)
+    except IOError:
```

Comment:
You can use `os.path.exists` to check whether the file exists before opening it.